### PR TITLE
Upgrade fontconfig 2.13.1

### PR DESCRIPTION
**Pull request by trueroad**
https://gitlab.com/lilypond/lilypond/-/issues/5287

Current GUB has fontconfig 2.12.6.
Its cache building speed is very slow, especially in an environment with CJK fonts, taking several minutes.
For example, in my Windows environment, it takes more than 5 minutes and 22 seconds.

Fontconfig 2.13 has improved the cache building speed.
I've created a pull request that upgrades fontconfig 2.13.1.

I've confirmed that lilypond-installer for 4 environments can be built.
And I've confirmed that the installers and installed LilyPond work fine for 3 environments except for darwin-x86.
In my Windows environment, the cache building takes only 7 seconds.

### CODE REVIEW

File: gub/specs/util-linux.py [hunk 1]
```diff
@@ -1,11 +1,52 @@
...
+    patches = ['util-linux-2.16-sysmacro.patch']
     configure_flags = (tools.AutoBuild.configure_flags
                 + ' --disable-tls'
                 + ' --disable-makeinstall-chown'
                 + ' SHELL=%(tools_prefix)s/bin/bash'
                 + ''' CFLAGS='-DLINE_MAX=1024' ''')
+
+class Util_linux (target.AutoBuild):
+    # util-linux-2.23 requires O_CLOEXEC but our glibc does not have it.
+    source = 'https://mirrors.edge.kernel.org/pub/linux/utils/util-linux/v2.22/util-linux-2.22.2.tar.xz'
```
Comment (lemzwerg) [inline gub/specs/util-linux.py:17]: Just curious: Do we really need both 'util-linux' versions 2.16 and 2.22.2?

**Comment by lemzwerg** [comment]
LGTM, thanks!

File: gub/specs/util-linux.py (hunk 1 above)
Comment (trueroad) [reply gub/specs/util-linux.py:17]: util-linux-2.22.2 could not compile for tools::util-linux.

Comment (lemzwerg) [reply gub/specs/util-linux.py:17]: OK.  Please add a remark!

Comment (trueroad) [reply gub/specs/util-linux.py:18]: done. Thanks.

**Review by hahnjo — CHANGES_REQUESTED** [review]
Doesn't compile for me from a fresh GUB:
```
Cannot update libtool: no such file: /code/LilyPond/gub/target/linux-64/root/usr/bin/libtool
```

I'm retrying with `libtool` added to `dependencies` of `class Util_linux`

**Review by hahnjo — CHANGES_REQUESTED** [review]
Then it goes on to complain in `tools::fontconfig` because that doesn't depend on `util-linux`. If I add that, it fails to build `tools::bash` because that references a global variable `no_patch` that is defined in `gub/specs/gzip.py`...

**Comment by trueroad** [comment]
Thank you for your review.

> Doesn't compile for me from a fresh GUB:
> 
> ```
> Cannot update libtool: no such file: /code/LilyPond/gub/target/linux-64/root/usr/bin/libtool
> ```
> 
> I'm retrying with `libtool` added to `dependencies` of `class Util_linux`

I've added the dependency.

> Then it goes on to complain in `tools::fontconfig` because that doesn't depend on `util-linux`. If I add that, it fails to build `tools::bash` because that references a global variable `no_patch` that is defined in `gub/specs/gzip.py`...

I've downgraded tools::fontconfig.

**Review by hahnjo — CHANGES_REQUESTED** [review]
Now it complains in `darwin-x86::fontconfig` that it cannot find `-luuid`. Not sure how I solved that one the last time...

**Review by hahnjo — APPROVED** [review]
> Now it complains in `darwin-x86::fontconfig` that it cannot find `-luuid`. Not sure how I solved that one the last time...

Okay, understood: I still had `tools::util-linux` around and `configure` found the `pkgconfig` file from `tools::`. Which is surprising, to say the least, but shouldn't happen if you religiously stick to `make lilypond`.

**Comment by hahnjo** [comment]
Hrm, there's now a new directory `sbin/` in the installed directory at a total size of 1.5 MB...

**Comment by trueroad** [comment]
> Hrm, there's now a new directory `sbin/` in the installed directory at a total size of 1.5 MB...

I've created another pull request that solves it.
#85

**Pull request merged**